watchdogutil/main.py

```diff
@@ -0,0 +1,116 @@
+#!/usr/bin/env python
+#
+# main.py
+#
+# Command-line utility for interacting with HW Watchdog in SONiC
+#
+
+try:
+    import sys
+    import os
+    import click
+    import syslog
+    import sonic_platform
+except ImportError as e:
+    raise ImportError("%s - required module not found" % str(e))
+
+VERSION = '2.0'
+
+SYSLOG_IDENTIFIER = "watchdogutil"
+
+WATCHDOG_LOAD_ERROR = -1
+CHASSIS_LOAD_ERROR = -2
+
+# Global platform-specific watchdog class instance
+platform_watchdog = None
+
+
+# ========================== Syslog wrappers ==========================
+
+
+def log_info(msg, also_print_to_console=False):
+    syslog.openlog(SYSLOG_IDENTIFIER)
+    syslog.syslog(syslog.LOG_INFO, msg)
+    syslog.closelog()
+
+    if also_print_to_console:
+        click.echo(msg)
+
+
+def log_warning(msg, also_print_to_console=False):
+    syslog.openlog(SYSLOG_IDENTIFIER)
+    syslog.syslog(syslog.LOG_WARNING, msg)
+    syslog.closelog()
+
+    if also_print_to_console:
+        click.echo(msg)
+
+
+def log_error(msg, also_print_to_console=False):
+    syslog.openlog(SYSLOG_IDENTIFIER)
+    syslog.syslog(syslog.LOG_ERR, msg)
+    syslog.closelog()
+
+    if also_print_to_console:
+        click.echo(msg)
+
+
+# ==================== Methods for initialization ====================
+
+# Loads platform specific watchdog module from source
+def load_platform_watchdog():
+    global platform_watchdog
+
+    platform = sonic_platform.platform.Platform()
+
+    chassis = platform.get_chassis()
+    if not chassis:
+        log_error("Failed to get chassis")
+        return CHASSIS_LOAD_ERROR
+
+    platform_watchdog = chassis.get_watchdog()
+    if not platform_watchdog:
+        log_error("Failed to get watchdog module")
+        return WATCHDOG_LOAD_ERROR
+
+    return 0
+
+
+# ==================== CLI commands and groups ====================
+
+
+# This is our main entrypoint - the main 'watchdogutil' command
+@click.group()
+def cli():
```

Comment:
Suggest to name the main group the same name as the utility, mainly for readability (i.e., `def watchdogutil`). This way, it's easier to understand the hierarchy when reading the code. You will also need to change all references in this file as well as change setup.py to read `watchdogutil = watchdogutil.main:watchdogutil`. I want to update all existing utilities which currently use "cli" in the future, also.